resolver/src/lookup_ip.rs

```diff
@@ -81,12 +80,68 @@ where
     names: Vec<Name>,
     strategy: LookupIpStrategy,
     options: DnsRequestOptions,
-    future: Box<Future<Item = Lookup, Error = ResolveError> + Send>,
+    query: Box<Future<Item = Lookup, Error = ResolveError> + Send>,
```

Comment:
Good name :)

Reply:
Thanks, I thought that was a bit clearer :) 

Though it did occur to me that there might potentially be a generic "retry until we run out of things to retry with" future combinator hidden in here, I couldn't think of an effective way to generalize it off the top of my head.